provider/azure.go

```diff
@@ -253,6 +259,14 @@ func (p *AzureProvider) zones(ctx context.Context) ([]dns.Zone, error) {
 	for zonesIterator.NotDone() {
 		zone := zonesIterator.Value()
 
+		if len(p.zoneNameFilter.filters) == 0 && !p.domainFilter.Match(*zone.Name) {
```

Comment:
wouldn't this be enough to check if len(p.zoneNameFilter.filters) == 0?

Reply:
fixed